hal/halx86/apic/halinit.c

```diff
@@ -25,16 +25,12 @@ HalpInitProcessor(
     IN ULONG ProcessorNumber,
     IN PLOADER_PARAMETER_BLOCK LoaderBlock)
 {
-#ifdef CONFIG_SMP
```

Comment:
Aha, yes, the same kind of problem I had before: the halinit.c was compiled only once (as part of those "static" component libraries) without CONFIG_SMP.

A nicer solution may be needed in the future, in case such files need to include non-trivial extra SMP-specific code within some of their existing functions.